assets/js/base/components/quantity-selector/index.js

```diff
@@ -47,17 +44,17 @@ const QuantitySelector = ( {
 					? event.key === 'ArrowUp'
 					: event.keyCode === UP;
 
-			if ( isArrowDown && currentValue > 0 ) {
+			if ( isArrowDown && canDecrease ) {
 				event.preventDefault();
-				setCurrentValue( currentValue - 1 );
+				onChange( quantity - 1 );
 			}
 
-			if ( isArrowUp ) {
+			if ( isArrowUp && canIncrease ) {
 				event.preventDefault();
-				setCurrentValue( currentValue + 1 );
+				onChange( quantity + 1 );
 			}
 		},
-		[ currentValue, setCurrentValue ]
+		[ quantity, onChange ]
```

Comment:
Either move `canIncrease` and `canDecrease` declarations inside the `useCallback` function scope or add as dependencies here. The current implementation risks having `canIncrease` and `canDecrease` being stale when the callback registered here is invoked.

Reply:
Good point, missed that - thanks for catching it.

Reply:
(Made these dependencies as they are used outside in the main component markup also.)